examples/gzip-stream/src/lib.rs

```diff
@@ -54,35 +54,21 @@ impl CompressStream {
     // After each call to `write` or `finish`, data may be written to the internal
     // buffer. This function copies the written data out and resets the buffer
     // to empty.
-    fn output(self) -> Result<Vec<u8>, CompressError> {
-        let mut guard = self.lock()?;
+    fn and_buffer(
+        mut cx: TaskContext,
+        // Return value from `cx.task(..)` closure
+        result: Result<Self, CompressError>,
+    ) -> JsResult<JsUint8Array> {
+        let stream = result.or_else(|err| cx.throw_error(err))?;
+        let mut guard = stream.lock().or_else(|err| cx.throw_error(err))?;
+
         let data = guard.get_mut();
-        let output = data.clone();
+        let output = JsUint8Array::from_slice(&mut cx, data)?;
```

Comment:
Since we were already cloning, we can get very similar ergonomics and performance by copying directly into the `Uint8Array` instead of using an `external`.